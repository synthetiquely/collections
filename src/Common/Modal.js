import React from 'react';
import glamorous from 'glamorous';

const Backdrop = glamorous.div({
  position: 'fixed',
  display: 'flex',
  width: '100%',
  height: '100%',
  top: '0px',
  left: '0px',
  zIndex: '1',
  backgroundColor: 'rgba(0,0,0,0.6)',
});

const Window = glamorous.div({
  width: '80vh',
  height: '80vh',
  zIndex: '2',
  padding: '30px',
  display: 'flex',
  margin: 'auto',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f6f5f3',
  '@media only screen and (max-width: 1024px) and (orientation: portrait)': {
    width: '80vw',
    height: '80vw',
  },
  '@media only screen and (max-width: 1024px) and (orientation: landscape)': {
    width: '80vh',
    height: '80vh',
  },
});

const CloseButton = glamorous.span({
  position: 'absolute',
  top: '-5px',
  right: '5px',
  color: '#aaa',
  fontSize: '28px',
  fontWeight: 'bold',
  cursor: 'pointer',
});

const Next = glamorous.div({
  zIndex: '3',
  position: 'absolute',
  top: '50%',
  right: '0',
  width: 'auto',
  marginTop: '-22px',
  padding: '10px',
  color: '#fff',
  fontWeight: 'bold',
  fontSize: '20px',
  transition: '0.6s ease',
  borderRadius: '0 3px 3px 0',
  cursor: 'pointer',
  ':hover,:active,:focus': {
    fontSize: '25px',
  },
  '@media only screen and (max-width: 500px) and (orientation: portrait)': {
    top: '90%',
    right: '50%',
    left: '50%',
    margin: '0',
    padding: '5px',
    transform: 'rotateZ(90deg)',
  },
});

const Prev = glamorous.div({
  zIndex: '3',
  position: 'absolute',
  top: '50%',
  width: 'auto',
  left: '0',
  marginTop: '-22px',
  padding: '10px',
  color: '#fff',
  fontWeight: 'bold',
  fontSize: '20px',
  transition: '0.6s ease',
  borderRadius: '3px 0 0 3px',
  cursor: 'pointer',
  ':hover,:active,:focus': {
    fontSize: '25px',
  },
  '@media only screen and (max-width: 500px) and (orientation: portrait)': {
    top: '0',
    left: '50%',
    margin: '0',
    padding: '5px',
    transform: 'rotateZ(90deg)',
  },
});

const Modal = (props) => {
  if (props.isOpen) {
    return (
      <React.Fragment>
        <Backdrop onClick={props.onClose}>
          <Window>
            <CloseButton onClick={props.onClose}>&times;</CloseButton>
            <Next onClick={props.onNext}>&#10095;</Next>
            <Prev onClick={props.onPrevious}>&#10094;</Prev>
            {props.children}
          </Window>
        </Backdrop>
      </React.Fragment>
    );
  }
  return null;
};

export default Modal;
