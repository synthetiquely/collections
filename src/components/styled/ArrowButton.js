import glamorous from 'glamorous';
import { css } from 'glamor';

const unfould = css.keyframes({
  '0%': { transform: 'scaleY(0) scaleX(0)' },
  '100%': { transform: 'scaleY(1) scaleX(1)' },
});

const ArrowButton = glamorous.button(
  {
    zIndex: '3',
    position: 'absolute',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '20px',
    color: '#fff',
    fontWeight: 'bold',
    fontSize: '20px',
    backgroundColor: 'rgba(0,0,0,.4)',
    border: 'none',
    cursor: 'pointer',
    userSelect: 'none',
    ':hover,:active,:focus': {
      fontSize: '25px',
      outline: 'none',
      backgroundColor: 'rgba(0,0,0,.5)',
    },
    animation: `${unfould} .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards`,
    '@media (orientation: portrait)': {
      width: '100%',
      height: 'auto',
    },
    '@media (orientation: landscape)': {
      width: 'auto',
      height: '100%',
    },
    '@media only screen and (max-width: 500px)': {
      padding: '10px',
    },
  },
  ({ direction }) => {
    if (direction === 'right') {
      return {
        right: '0',
        borderRadius: '0 3px 3px 0',
        '@media only screen and (max-width: 1750px) and (orientation: portrait)': {
          bottom: '0',
          margin: '0',
        },
      };
    } else if (direction === 'left') {
      return {
        left: '0',
        borderRadius: '3px 0 0 3px',
        '@media only screen and (max-width: 1750px) and (orientation: portrait)': {
          top: '0',
          margin: '0',
        },
      };
    }
    return {};
  },
);

export default ArrowButton;
