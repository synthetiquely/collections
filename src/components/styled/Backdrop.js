import glamorous from 'glamorous';

const Backdrop = glamorous.div({
  position: 'fixed',
  display: 'flex',
  width: '100%',
  height: '100%',
  top: '0px',
  left: '0px',
  zIndex: '1',
  backgroundColor: 'rgba(0,0,0,0.6)',
});

export default Backdrop;
