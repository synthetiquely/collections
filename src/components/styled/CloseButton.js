import glamorous from 'glamorous';

const CloseButton = glamorous.button({
  zIndex: '4',
  position: 'absolute',
  top: '0',
  right: '10px',
  color: '#aaa',
  fontSize: '28px',
  fontWeight: 'bold',
  backgroundColor: 'transparent',
  border: 'none',
  cursor: 'pointer',
  userSelect: 'none',
});

export default CloseButton;
