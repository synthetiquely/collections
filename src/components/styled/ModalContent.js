import glamorous from 'glamorous';
import { css } from 'glamor';

const blowUp = css.keyframes({
  '0%': { transform: 'scale(0)' },
  '100%': { transform: 'scale(1)' },
});

const ModalContent = glamorous.div(
  {
    willChange: 'backgroundColor',
    transition: '0.5s ease',
    animation: `${blowUp} .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards`,
  },
  ({ color }) => {
    if (color) {
      return {
        backgroundColor: color,
      };
    }
    return {
      backgroundColor: 'transparent',
    };
  },
);

export default ModalContent;
