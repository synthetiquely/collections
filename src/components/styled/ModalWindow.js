import glamorous from 'glamorous';

const ModalWindow = glamorous.div({
  width: '80vh',
  height: '80vh',
  zIndex: '2',
  display: 'flex',
  margin: 'auto',
  justifyContent: 'center',
  alignItems: 'center',
  '@media only screen and (max-width: 1024px) and (orientation: portrait)': {
    width: '90vw',
    height: '90vw',
  },
  '@media only screen and (max-width: 1024px) and (orientation: landscape)': {
    width: '90vh',
    height: '90vh',
  },
  '@media only screen and (max-width: 1024px)': {
    padding: '10px',
  },
});

export default ModalWindow;
