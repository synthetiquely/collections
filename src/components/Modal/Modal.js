import React from 'react';
import { DESTINATION_NEXT, DESTINATION_PREVIOUS } from '../../constants';
import Backdrop from '../styled/Backdrop';
import ModalWindow from '../styled/ModalWindow';
import ModalContent from '../styled/ModalContent';
import ArrowButton from '../styled/ArrowButton';
import CloseButton from '../styled/CloseButton';
import Icon from '../styled/Icon';

const Modal = props => (
  <Backdrop>
    <ModalWindow id="preview-modal">
      <CloseButton id="preview-close-button" onClick={props.onClose}>
        <Icon size="lg">&times;</Icon>
      </CloseButton>
      <ArrowButton
        id="preview-right-arrow"
        direction="right"
        onClick={() => props.onChangeSelected(DESTINATION_NEXT)}
      >
        <Icon rotatable size="md">
          &#10095;
        </Icon>
      </ArrowButton>
      <ArrowButton
        id="preview-left-arrow"
        direction="left"
        onClick={() => props.onChangeSelected(DESTINATION_PREVIOUS)}
      >
        <Icon rotatable size="md">
          &#10094;
        </Icon>
      </ArrowButton>
      <ModalContent
        onMouseEnter={props.onMouseEnter}
        onMouseLeave={props.onMouseLeave}
        color={props.bgColor}
      >
        {props.children}
      </ModalContent>
    </ModalWindow>
  </Backdrop>
);

export default Modal;
