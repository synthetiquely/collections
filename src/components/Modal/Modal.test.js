import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Modal from './Modal';
import Backdrop from '../styled/Backdrop';
import ModalWindow from '../styled/ModalWindow';
import ModalContent from '../styled/ModalContent';
import ArrowButton from '../styled/ArrowButton';
import CloseButton from '../styled/CloseButton';

describe('Modal Component', () => {
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<Modal />);
    const tree = renderer.create(<Modal />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
    expect(wrapper.find(Backdrop)).toHaveLength(1);
    expect(wrapper.find(ModalWindow)).toHaveLength(1);
    expect(wrapper.find(ModalContent)).toHaveLength(1);
    expect(wrapper.find(ArrowButton)).toHaveLength(2);
    expect(wrapper.find(CloseButton)).toHaveLength(1);
  });
});
