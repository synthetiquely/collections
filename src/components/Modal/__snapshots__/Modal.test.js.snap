// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render and match its own snapshot 1`] = `
<div
  className="css-3hsllp"
>
  <div
    className="css-13la5a5"
    id="preview-modal"
  >
    <button
      className="css-hcjupp"
      id="preview-close-button"
      onClick={undefined}
    >
      <span
        className="css-16yzkgv"
      >
        ×
      </span>
    </button>
    <button
      className="css-10e1v19"
      id="preview-right-arrow"
      onClick={[Function]}
    >
      <span
        className="css-937lsz"
      >
        ❯
      </span>
    </button>
    <button
      className="css-1cvb4bz"
      id="preview-left-arrow"
      onClick={[Function]}
    >
      <span
        className="css-937lsz"
      >
        ❮
      </span>
    </button>
    <div
      className="css-1web6nd"
      onMouseEnter={undefined}
      onMouseLeave={undefined}
    />
  </div>
</div>
`;
